aws/sdk/examples/config/src/bin/enable-config.rs

```diff
@@ -0,0 +1,191 @@
+    #[structopt(default_value = "AWS::DynamoDB::Table", short, long)]
+    type_: String,
+
+    /// Whether to display additional information.
+    #[structopt(short, long)]
+    verbose: bool,
+}
+
+/// Enables AWS Config for a resource type, in the Region.
+///
+/// # Arguments
+///
+/// * `-b BUCKET` - The name of the Amazon bucket to which AWS Config delivers configuration snapshots and configuration history files.
+/// * `-i IAM-ARN` - The ARN of the IAM role that used to describe the AWS resources associated with the account.
+/// * `-k KMS-ARN` - The ARN of the KMS key that used to encrypt the data in the bucket.
+/// * `-p PREFIX` - The prefix for the bucket.
+/// * `-s SNS-ARN` - The ARN of the Amazon SNS topic to which AWS Config sends notifications about configuration changes.
+/// * `[-t TYPE]` - The type of resource for AWS Config to support.
+///   If not supplied, defaults to `AWS::DynamoDB::Table` (DynamoDB tables).
+/// * `[-n NAME]` - The name of the configuration.
+///   If not supplied, defaults to `default`.
+/// * `[-r REGION]` - The Region in which the client is created.
+///   If not supplied, uses the value of the **AWS_REGION** environment variable.
+///   If the environment variable is not set, defaults to **us-west-2**.
+/// * `[-v]` - Whether to display information.
+/// Need: s3 key prefix AND kms key
+#[tokio::main]
+async fn main() -> Result<(), Error> {
+    tracing_subscriber::fmt::init();
+    let Opt {
+        region,
+        bucket,
+        iam_arn,
+        kms_arn,
+        name,
+        prefix,
+        sns_arn,
+        type_,
+        verbose,
+    } = Opt::from_args();
+
+    let region_provider = RegionProviderChain::first_try(region.map(Region::new))
+        .or_default_provider()
+        .or_else(Region::new("us-west-2"));
+
+    println!();
+
+    if verbose {
+        println!("Config client version:          {}", PKG_VERSION);
+        println!(
+            "Region:                {}",
+            region_provider.region().await.unwrap().as_ref()
+        );
+        println!("Resource type:                  {}", type_);
+        println!("Config (delivery channel) name: {}", name);
+        println!("Bucket:                         {}", bucket);
+        println!("Prefix:                         {}", prefix);
+        println!("SNS ARN:                        {}", sns_arn);
+        println!("IAM ARN:                        {}", iam_arn);
+        println!("KMS ARN:                        {}", kms_arn);
+        println!();
+    }
+
+    let shared_config = aws_config::from_env().region(region_provider).load().await;
+    let client = Client::new(&shared_config);
+
+    // If we already have a configuration recorder in the Region, we cannot create another.
+    let resp = client.describe_configuration_recorders().send().await?;
+
+    let recorders = resp.configuration_recorders.unwrap_or_default();
+
+    let num_recorders = recorders.len();
+
+    if num_recorders != 0 {
```

Comment:
This can be simplified:
```suggestion
    if !recorders.is_empty() {
```

Reply:
Fixed